src/Illuminate/Console/Application.php

```diff
@@ -95,7 +96,7 @@ public static function phpBinary()
      */
     public static function artisanBinary()
     {
-        return ProcessUtils::escapeArgument(defined('ARTISAN_BINARY') ? ARTISAN_BINARY : 'artisan');
+        return ProcessUtils::escapeArgument(artisan_binary());
```

Comment:
I don't feel like we need to introduce `artisan_binary()` helper since calling `Illuminate\Console\Application::artisanBinary()` would always yield the correct value.

Reply:
```suggestion
        return ProcessUtils::escapeArgument(defined('ARTISAN_BINARY') ? ARTISAN_BINARY : 'artisan');
```

Reply:
I feel like this method should return without quote in Laravel 12 and let method that needs it to explicit espace the argument.

Escaping argument is only needed when we providing `string` instead of `array` to Symfony Process.

Reply:
`artisan_binary()` also returns without quotes, the change on this line doesn't change the output from `defined('ARTISAN_BINARY') ? ARTISAN_BINARY : 'artisan'`

I agree with you that there's further cleanup possible for v12